EIPS/eip-uncouple-payload.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Uncouple execution payload from beacon block
+description: Separates the execution payload from beacon block to independently transmit them
+author: Gajinder Singh <@g11tech>
+discussions-to: https://ethereum-magicians.org/t/uncouple-execution-payload-from-beacon-block/23029
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-03-01
+---
+
+## Abstract
+
+Currently the beacon block in ethereum consensus embeds the transactions in `ExecutionPayload` field of `BeaconBlockBody`. This EIP proposes to replace `ExecutionPayload` with `ExecutionPayloadHeader` in `BeaconBlockBody` and to independently transmit `ExecutionPayloadWithInclusionProof`.
+
+However this EIP makes no change to the block import mechanism except that block availability also now includes waiting for the availability of `ExecutionPayloadWithInclusionProof` and this is where its different (and simpler) from proposals like ePBS.
+
+But this availability requirement can infact be restricted to `gossip` import while allowing optimistic syncing of ELs on checkpoint/range sync as EL can pull full blocks from their peers in optimistic sync as they do now.
+
+## Motivation
+
+Ethereum protocol has an ambitious goal to grow the `gasLimit` of the execution payloads (possibly by 10X). This introduces bigger and bigger messages in the networking and  block processing pipeline of the CL clients leading to following issues:
+
+1. Latency for the arrival of Beacon Block increases and make it more and more dependent on the bandwidth resources available to the running node.
+2. The greater number/size of `transactions` directly increase the merkelization compute time directly affecting the import time of the block.
+
+We know from timing games that the block import latency greatly affects a client's performance to make correct head attestations. With this EIP, block transmission and import pipeline will be unchocked and allows for greater flexibility in reception of huge `ExecutionPayloadWithInclusionProof` while the beacon block can undergo processing. 
+
+Infact EL clients can also independently participate in the propagation/pull of the execution blocks. That mechanism however can be independently developed and is not the part of this EIP.
+
+Other side benefits of this EIP are that 
```

Comment:
I think the minor benefit is that there is no need to maintain `BlindedBeaconBlock` and `BlindedBeaconBlockBody` container since everything will be blinded after this EIP so there will only be one type of beacon block

Reply:
thanks, added to the benefits :)